src/fal/planner/plan.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from typing import Iterator
+
+import networkx as nx
+
+
+def _find_subgraphs(graph: nx.DiGraph) -> Iterator[list[str]]:
+    # Initially topologically sort the graph, and split it
+    # into chunks by using the critical node approach (post-hooks
+    # are also a thing now).
```

Comment:
I think the 
> (post-hooks are also a thing now)

could stay there after post-hooks are a normal thing

Maybe update with a list of things we handle

```
- post-hooks
- after-scripts
- before-scripts
- ...
```

or don't put this part of the comment